### PR TITLE
Add description column to Cluster Settings Global Configuration table

### PR DESCRIPTION
**Pull request by rhamilto**
https://jira.coreos.com/browse/CONSOLE-1456

Screen Shot 2019-07-31 at 10 20 53 AM

### CODE REVIEW

**Comment by cshinn** [comment]
@rhamilto Looks like there's a recently merged docs PR that I believe is related. Are these the same ones that are appearing from the API server right now? https://github.com/openshift/openshift-docs/pull/15066

**Comment by rhamilto** [comment]
> Are these the same descriptions that are appearing from the API server right now?

No because no descriptions are appearing as a result of a [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1712317).  What should be appearing looks like https://github.com/openshift/api/blob/master/config/v1/types_apiserver.go#L11, and those need to be rewritten as they're all of that format, which is not exactly helpful.

File: frontend/public/components/cluster-settings/global-config.tsx [hunk 1]
```diff
@@ -3,21 +3,29 @@ import * as _ from 'lodash-es';
...
 
+const allDefinitions: SwaggerDefinitions = getStoredSwagger();
+
 const globalConfigListStateToProps = ({k8s}) => ({
   configResources: k8s.getIn(['RESOURCES', 'configResources']),
 });
 
 const ItemRow = ({item}) => {
   const resourceLink = resourcePathFromModel(item.model, item.metadata.name, item.metadata.namespace);
+  const itemPath = [Object.keys(allDefinitions).find(key => key.endsWith(`openshift.config.${item.model.apiVersion}.${item.kind}`))];
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/global-config.tsx]: You should be able to use the helper from #1676

Comment (rhamilto) [reply frontend/public/components/cluster-settings/global-config.tsx]: Nice!  Updated.

**Comment by rhamilto** [comment]
This is unblocked, but the unblocking came at feature freeze leaving no time for us to test and tag.  Will have to wait until 4.3?

**Comment by spadgett** [comment]
This is borderline a usability bug, although we probably want better descriptions before we go ahead with the change

**Comment by rhamilto** [comment]
@spadgett, should you decide this should go in, it's ready.

File: frontend/public/components/cluster-settings/global-config.tsx [hunk 2]
```diff
@@ -39,11 +41,16 @@ const ItemRow = ({item}) => {
     viewAPIExplorerMenuItem(item.kind, apiExplorerLink),
     ...(item.kind === 'OAuth') ? oauthMenuItems : [],
   ];
+  const itemKey = getDefinitionKey(item.model, allDefinitions);
+  const itemDefinition: SwaggerDefinition = _.get(allDefinitions, itemKey) || {};
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/global-config.tsx]: This should use `getResourceDescription`

https://github.com/openshift/console/blob/master/frontend/public/module/k8s/swagger.ts#L51-L58

**Pull request merged**